### PR TITLE
Google Authentication for Android applications

### PR DESCRIPTION
**Pull request by BrianaNzivu**
Created an index.md file for Google authentication for Android applications.

The article illustrates how to integrate google authentication in android applications.

Extra images included are;
empty.png
finalone.jpg
finaltwo.jpg
finalthree.jpg
name.png
newproject.png
xml.png

### CODE REVIEW

File: articles/google-authentication-for-android/index.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+Let's run the app.
+
+
+![GoogleSignUp](/engineering-education/google-authentication-for-android/finalone.jpg)
+
+![GoogleSignUp](/engineering-education/google-authentication-for-android/finaltwo.jpg)
+
+![GoogleSignUp](/engineering-education/google-authentication-for-android/finalthree.jpg)
+
+You can get the source code [here](https://github.com/BrianaNzivu/googlesignup/blob/master/app/src/main/java/com/example/googlesignup/MainActivity.java).
```
Comment (LinusMuema) [inline articles/google-authentication-for-android/index.md:250]: You could provide the link to the entire repo. Not just the activity file.

**Comment by LinusMuema** [comment]
Hi @BrianaNzivu 
This is an awesome article, quite informative. I would suggest you take a look at the markdown headlines i.e using `###` etc. I have also noticed that you do not have an author's directory. In this same PR, just create a directory with your name and add the `avatar`  and your description. Something like [this](https://github.com/section-io/engineering-education/tree/master/authors/linus-muema)

**Comment by LinusMuema** [comment]
Hey @hectorkambow 
You can take a look at this article before I get into the tech review.

**Comment by BrianaNzivu** [comment]
Good morning Linus.
Let me work on it right now.

Kind Regards.

**Comment by BrianaNzivu** [comment]
Hey @LinusMuema .
I made the changes.
Kindly review.

Kind Regards.

**Comment by LinusMuema** [comment]
Hi @BrianaNzivu 
I have done a technical review and I have some pointers. I have noticed that you have used the ` .requestIdToken(getString(R.string.default_web_client_id))` which returns an ApiException of code 10. This can be confusing to new readers. You can give them a brief guide on how to get a valid clientId or just a link to resources that can help them out. Also, once an exception occurs, you have directed them to proceed to the `Home` activity. I would suggest you just show a toast or snackbar without proceeding since the user is not actually authenticated. More of an error handling mechanism.

**Comment by BrianaNzivu** [comment]
Hi @linusmuema.
The token is normally automatically added once you build and run the app.
What I can do is maybe add that statement in the article just to make sure
it's not confusing for the readers.

I never noticed the last error... Let me fix it and push the changes.

Regards.

**Comment by LinusMuema** [comment]
I have actually cloned the repo and run it. So maybe it's my IDE that has the issue. Just include that in the article and I'll check more into that.

**Comment by BrianaNzivu** [comment]
Hey @LinusMuema .
Made the corrections.
Regards.

**Comment by LinusMuema** [comment]
Hi @hectorkambow could you please give this article a final review.

**Comment by hectorkambow** [comment]
@BrianaNzivu  👋  Good afternoon Briana. 

Thank you for submitting your "Google Authentication for Android applications" article. 
Upon running your article through our 3rd party plagiarism check it seemed to raise a few red flags. 

If you could see the attached PDF and revisit the article to change any underline sentences. 

Thank you 🚀 
[Uploading Google Authentication for Android applications #400.pdf…]()

**Comment by BrianaNzivu** [comment]
Good morning @hectorkambow .The link seems to be redirecting me to the same page.

**Comment by BrianaNzivu** [comment]
Hello @LinusMuema .
Kindly confirm if the draft is ready for technical review.

Kind Regards
.

**Comment by LinusMuema** [comment]
I already did the technical review. It was only the plagiarism check that was holding this back. @hectorkambow can confirm if this is ready for publishing

**Comment by BrianaNzivu** [comment]
Hello @hectorkambow .
I made the changes required.
Kindly confirm if the draft is ready for publishing.

**Pull request merged**